src/gui/editorwidgets/core/qgseditorwidgetfactory.cpp

```diff
@@ -123,3 +123,27 @@ unsigned int QgsEditorWidgetFactory::fieldScore( const QgsVectorLayer* vl, int f
   return 5;
 }
 
+void QgsEditorWidgetFactory::config2xml( const QgsEditorWidgetConfig& config, QDomElement& configElement, const QString& fieldName )
+{
+  const QVariant value = config.value( fieldName );
+  if ( value.isValid() )
+  {
+    if ( value.type() == QVariant::Bool )
+    {
+      configElement.setAttribute( fieldName, value.toBool() ? "1" : "0" );
+    }
+    else
+    {
+      configElement.setAttribute( fieldName, value.toString() );
+    }
+  }
+}
+
+void QgsEditorWidgetFactory::xml2config( const QDomElement& configElement, QgsEditorWidgetConfig& config, const QString& fieldName )
+{
+  const QString value = configElement.attribute( fieldName );
+  if ( !value.isNull() )
+  {
+    config.insert( fieldName, value );
+  }
```

Comment:
Funny that the example you are mentioning (DocumentViewerWidth) is not used :-)

The config is QVariantMap. So storing it as a string or directly as a an int won't change much. Where it is used, we have to do the `toInt()` anyway.

I could add a `QVariant::Type` parameter to `xml2config` and call `QVariant::convert` in there, but I really don't see the point...